qemu/tests/uefishell.py

```diff
@@ -0,0 +1,152 @@
+import os
+import re
+import logging
+
+from avocado.utils import process
+
+from virttest import storage
+from virttest import data_dir
+from virttest import utils_misc
+from virttest import env_process
+
+
+class UEFIShellTest(object):
+    """
+    Provide basic functions for uefishell test. To use UefiShell.iso
+    which is provided by ovmf package. Boot your VM with this as
+    the CD-ROM image and it should boot into the UEFI shell.
+    """
+
+    def __init__(self, test, params, env):
+        """
+        Init the default values for UEFIShell object
+        :param test: Kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        """
+        self.test = test
+        self.params = params
+        self.env = env
+        self.session = None
+
+    def setup(self, under_fs0):
+        """
+        Pre-process for uefishell environment, launch vm from the
+        UefiShell.iso, and setup a serial session for uefishell
+
+        :param under_fs0: most uefi command executed under fs0:\
+        """
+        params = self.params
+        ovmf_path = params["ovmf_path"]
+        uefishell_filename = "UefiShell.iso"
+        uefishell_dst_path = "images/%s" % uefishell_filename
+        params["cdroms"] = "%s %s" % (
+            params["cdroms"], "uefishell")
+        params["cdrom_uefishell"] = uefishell_dst_path
+        uefishell_src_path = utils_misc.get_path(
+            ovmf_path, uefishell_filename)
+        uefishell_dst_path = utils_misc.get_path(
+            data_dir.get_data_dir(), uefishell_dst_path)
+        if not os.path.exists(uefishell_dst_path):
+            cp_command = "cp -f %s %s" % (
+                uefishell_src_path, uefishell_dst_path)
+            process.system(cp_command)
+        params["bootindex_uefishell"] = "0"
```

Comment:
If you specify bootindex=0 here, there will be following errors:
output: 'qemu-kvm: -device ide-cd,id=uefishell,drive=drive_uefishell,bootindex=0,bus=ide.0,unit=0: The bootindex 0 has already been used'

So you should also specify bootindex for image1.

Reply:
Bootindex errors still exist in latest code, pls check it.